WordPress/gutenberg
# Block Editor: Move component styles needed for iframes to content styles

Follow-up on: #44298

## What?
This PR moves some of the block editor component styles to be applied to the iframe editor instance. This will ensure that the styles of some components are applied correctly.

## Why?
In #44298, some component styles in the block editor package are now split and loaded into the iframe editor instance.

However, I think the following component styles should be loaded into the iframe: 

- ~`block-variation-picker`: Used for variation pickers like the column block~ (Fixed by #47109)
- `warning`: Used for UI to recover when a block is broken
- `block-icon`: Used for the title icon of the placeholder component

## How?
I moved ~`block-variation-picker` and~ `warning` from `style.css` to `content.css`. For `block-icon`, I copied `style.css` and created `content.css` since it is also used in the sidebar.

## Testing Instructions

Paste the following markup into the editor and confirm that the styles are applied correctly.

<details>
<summary>Test Data</summary>

```html

<p>Column Block requires <strong>Block Variation Picker</strong> style:</p>



<div class="wp-block-columns"></div>



<p>Blocks with placeholder icons require <strong>Block Icon</strong> style:</p>





<p>Invalid Blocks require <strong>Warning</strong> style.</p>



<p>Lorem Ipsum</p>

```
</details>


## Screenshots or screencast 

| Before | After |
| --- | --- |
| ![before](https://user-images.githubusercontent.com/54422211/212004408-8a1b9e43-81fa-43d8-bd67-7a774ac21158.png) | ![after](https://user-images.githubusercontent.com/54422211/212004450-869ed274-b317-4548-81e4-33c770e697e9.png) |
